packages/cli/generators/model/index.js

```diff
@@ -420,6 +420,11 @@ module.exports = class ModelGenerator extends ArtifactGenerator {
       ];
 
       const answers = await this.prompt(prompts);
+
+      if (answers.id) {
+        // 'generated' makes sure id is defined, especially for database like MySQL
+        answers['generated'] = true;
```

Comment:
Should we have a prompt to ask if the `id is automatically generated by the database`?

Reply:
+1 to add a prompt. The topic is more complex though because of the model-level setting `forceId`.

Let's land this PR as an incremental improvement and open a new issue to investigate what changes are needed to support models where the PK is provided by clients (not generated by the DB).